consensus/ibft/src/test/java/org/hyperledger/besu/consensus/ibft/IbftProtocolScheduleTest.java

```diff
@@ -0,0 +1,114 @@
+
+import java.math.BigInteger;
+import java.util.Collection;
+import java.util.List;
+
+import org.junit.Test;
+
+public class IbftProtocolScheduleTest {
+  private final BftExtraDataCodec bftExtraDataCodec = mock(BftExtraDataCodec.class);
+  private final NodeKey proposerNodeKey = NodeKeyUtils.generate();
+  private final Address proposerAddress = Util.publicKeyToAddress(proposerNodeKey.getPublicKey());
+  private final List<Address> validators = singletonList(proposerAddress);
+
+  @Test
+  public void blockModeTransitionsCreatesBlockModeHeaderValidators() {
+    final MutableBftConfigOptions arbitraryTransition =
+        new MutableBftConfigOptions(JsonQbftConfigOptions.DEFAULT);
+    arbitraryTransition.setBlockRewardWei(BigInteger.ONE);
+
+    final BlockHeader parentHeader =
+        IbftBlockHeaderUtils.createPresetHeaderBuilder(1, proposerNodeKey, validators, null)
+            .buildHeader();
+    final BlockHeader blockHeader =
+        IbftBlockHeaderUtils.createPresetHeaderBuilder(2, proposerNodeKey, validators, parentHeader)
+            .buildHeader();
+
+    final ProtocolSchedule schedule =
+        createProtocolSchedule(
+            JsonGenesisConfigOptions.fromJsonObject(JsonUtil.createEmptyObjectNode()),
+            new BftForkSpec<>(0, JsonQbftConfigOptions.DEFAULT),
+            List.of(
+                new BftForkSpec<>(1, arbitraryTransition),
+                new BftForkSpec<>(2, JsonQbftConfigOptions.DEFAULT)));
+    assertThat(schedule.streamMilestoneBlocks().count()).isEqualTo(3);
+    assertThat(validateHeader(schedule, validators, parentHeader, blockHeader, 0)).isTrue();
+    assertThat(validateHeader(schedule, validators, parentHeader, blockHeader, 1)).isTrue();
+    assertThat(validateHeader(schedule, validators, parentHeader, blockHeader, 2)).isTrue();
+  }
+
+  private ProtocolSchedule createProtocolSchedule(
+      final GenesisConfigOptions genesisConfig,
+      final BftForkSpec<BftConfigOptions> genesisFork,
+      final List<BftForkSpec<BftConfigOptions>> forks) {
+    return IbftProtocolSchedule.create(
+        genesisConfig,
+        new BftForksSchedule<>(genesisFork, forks),
+        PrivacyParameters.DEFAULT,
+        false,
+        bftExtraDataCodec,
+        EvmConfiguration.DEFAULT);
+  }
+
+  private boolean validateHeader(
+      final ProtocolSchedule schedule,
+      final List<Address> validators,
+      final BlockHeader parentHeader,
+      final BlockHeader blockHeader,
+      final int block) {
+    return schedule
+        .getByBlockNumber(block)
+        .getBlockHeaderValidator()
+        .validateHeader(
+            blockHeader, parentHeader, protocolContext(validators), HeaderValidationMode.LIGHT);
+  }
+
+  private ProtocolContext protocolContext(final Collection<Address> validators) {
+    return new ProtocolContext(
+        null,
+        null,
+        setupContextWithBftExtraDataEncoder(
+            BftContext.class, validators, new IbftExtraDataCodec()));
```

Comment:
Q: I noticed that you created a mock `bftExtraDataCodec` earlier. Do you really want a real instance here? Or maybe it doesn't matter?

Reply:
doesn't matter that much but changing it back to a mock